hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java

```diff
@@ -94,8 +94,9 @@ protected Map<EntityIDPair, Long> fungibleChangesFrom(
             final var relAcctId = modifiedRel.accountId();
             final var relTokenId = modifiedRel.tokenId();
             final var token = tokenStore.get(relTokenId);
-            // Add this to fungible token transfer list only if this token is a fungible token
-            if (!token.tokenType().equals(TokenType.FUNGIBLE_COMMON)) {
+            // Add this to fungible token transfer list only if this token is a fungible token or if the token is
+            // deleted
+            if (!token.deleted() && !token.tokenType().equals(TokenType.FUNGIBLE_COMMON)) {
```

Comment:
Why do we need this check ?  `!token.deleted()`

Reply:
This hapi test `dissociateHasExpectedSemanticsForDeletedTokens` requires the receipt for the `uniqDissoc` dissociate transaction to contain the a `tokenTransferList` like this:

```
tokenTransferLists {
  token {
    tokenNum: 1111
  }
  transfers {
    accountID {
      accountNum: 1109
    }
    amount: -3
  }
}
```

Which is built in `fungibleChangesFrom` if the `continue` statement is skipped. I think it does make more sense to add the transfers as `nftTransfers` since the token is a `NON_FUNGIBLE_UNIQUE`, but then the behaviour would differ from the mono service.

Reply:
I think it's very concerning to have record-construction logic depend on **anything** but the difference between the values returned by the `getOriginalValue()` and `get()` methods on the stores. (In this case, the `WritableTokenRelationStore`).

So even if this fixes a `@HapiTest`, I am not comfortable committing it. What do you think @Neeharika-Sompalli ?

Reply:
Yes agreed @tinker-michaelj

Reply:
I removed the `token.deleted()` condition and disabled the HapiTest `dissociateHasExpectedSemanticsForDeletedTokens` as I am not sure how to properly fix it.